src/db.rs

```diff
@@ -469,6 +479,55 @@ impl Database {
             .await
     }
 
+    /// Runs a database-level command and returns a cursor to the response.
+    pub async fn run_cursor_command(
+        &self,
+        command: Document,
+        options: RunCursorCommandOptions,
+    ) -> Result<Cursor<Document>> {
+        let rcc = RunCommand::new(self.name().to_string(), command, None, None)?;
+        let rc_command = RunCursorCommand {
+            run_command: rcc,
+            options,
+        };
+        let client = self.client();
+        client.execute_cursor_operation(rc_command).await
+    }
+
+    /// Runs a database-level command and returns a cursor to the response.
+    pub async fn run_cursor_command_with_session(
+        &self,
+        command: Document,
+        options: RunCursorCommandOptions,
+        session: &mut ClientSession,
+    ) -> Result<SessionCursor<Document>> {
+        let mut selection_criteria = options.read_preference
+            .clone()
+            .map(SelectionCriteria::ReadPreference);        
+        match session.transaction.state {
+            TransactionState::Starting | TransactionState::InProgress => {
+                selection_criteria = match selection_criteria {
+                    Some(selection_criteria) => Some(selection_criteria),
+                    None => {
+                        if let Some(ref options) = session.transaction.options {
+                            options.selection_criteria.clone()
+                        } else {
+                            None
+                        }
+                    }
+                };     
+            }
+            _ => {}
+        }
```

Comment:
I think you should be able to use the `resolve_selection_criteria_with_session` macro here ([e.g.](https://docs.rs/mongodb/latest/src/mongodb/coll/mod.rs.html#342-358)).

Reply:
It seems like the option types don't match up. Do we want to switch to that type or use the one we are currently using.
![image](https://github.com/mongodb/mongo-rust-driver/assets/67125579/274a92cf-2e64-4865-b712-81f92440c7d6)

Reply:
![image](https://github.com/mongodb/mongo-rust-driver/assets/67125579/0d511be7-a274-4569-a75a-9fbb5e91a9c9)
seems like the macro won't work out since the options types don't match up.